README.md

```diff
@@ -1,2 +1,162 @@
+```C#
+using Ockham.Data;
+
+     T Convert.To<T>(object);
+object Convert.To(object, Type);
+```
+
+This intuitively works for `DBNull`, `Guid`, `Timespan`, `bool`, `Nullable<T>`, and all enums (even nullables of enums!), and will automatically use `IConvertible` and/or user defined conversion operators where applicable. 
+
+Subtleties of these conversions can be controlled with the `ConvertOptions` class, described in more detail below:
+
+```C#
+object Convert.To(object, Type, ConvertOptions);
+     T Convert.To<T>(object, ConvertOptions);
+```
+
+Several type-specific overloads are also defined for convenience, which use the `ConvertOptions.Default` settings:
+
+```C#
+  bool Convert.ToBool(object)
+  Guid Convert.ToGuid(object)
+   int Convert.ToInt(object)
+  long Convert.ToLong(object)
+string Convert.ToString(object)
+... etc
+```
+
+If desired, conversion failures can be ignored (it is left to the user to determine when this is wise), with or without a default fallback value:
+
+```C#
+object Force.To(object value, Type targetType);
+object Force.To(object value, Type targetType, object defaultValue);
+     T Force.To<T>(object value);
+     T Force.To<T>(object value, T defaultValue);
+```
+ 
+Finally, methods are provided to convert to and from `DBNull`, or detect a "null" value (with the user specifying which things should be considered null):
+
+```C#
+object Convert.ToNull(object)  // Converts DBNull to null, leaves other values as-is
+object Convert.ToDBNull(object)  // Converts null to DBNull, leaves other values as-is
+
+// More control over what is considered null:
+object Convert.ToNull(object value, ConvertOptions options) 
+object Convert.ToNull(object value, bool emptyStringAsNull) 
+object Convert.ToDBNull(object value, ConvertOptions options) 
+object Convert.ToDBNull(object value, bool emptyStringAsNull) 
+
+// Or to simply inspect:
+bool Value.IsNull(object value)     // True if null or DBNull
+bool Value.IsNull(object value, bool emptyStringAsNull)
+bool Value.IsNull(object value, ConvertOptions options)
+```
+
+Similar functions are provided on the `Converter` class, where they use whatever options have been configured on the converter instance:
+
+```C#
+// Where 'other empty value' is determined by the converter's ConvertOptions 
+converter.ToNull(value)    // Converts DBNull or other empty value to null
+converter.ToDBNull(value)  // Converts null or other empty value to DBNull
+converter.IsNull(value)    // Detects null, DBNull, or other empty value
+```
+
+## Customization
+ 
+A specific set of conversion options can be reused more easily with an instance of the `Converter` class
+
+```C#
+var options = ConvertOptionsBuilder.Create()
+  .WithEnumOptions(undefinedNames: UndefinedValueOption.Ignore)
+  .WithStringOptions(emptyStringOptions: EmptyStringConvertOptions.WhitespaceAsNull, allowHex: true)
+  .WithTrueStrings(bool.TrueString, 't', 'x', 'y')
+  .WithFalseStrings(bool.FalseString, 'f', '', 'x')
+  .Build();
```

Comment:
Can we try a property called `options` or something?  `Build()` has a "scary" feeling for some reason :)

Reply:
Agreed. It makes it sound like there's some magic or heavy lifting, when it should just be a cheap assignment or property get.